web/packages/teleterm/src/ui/appContext.ts

```diff
@@ -176,8 +176,9 @@ export default class AppContext implements IAppContext {
 
     this.subscribeToDeepLinkLaunch();
     this.notifyMainProcessAboutClusterListChanges();
-    this.clustersService.syncGatewaysAndCatchErrors();
+    void this.clustersService.syncGatewaysAndCatchErrors();
     await this.clustersService.syncRootClustersAndCatchErrors();
+    this.workspacesService.restorePersistedState();
```

Comment:
If I close the app while I'm in workspace A with a valid cert and open it again through a deep link to workspace B, I can close the login modal and continue to use workspace A without making a decision wrt restored state. The user should be forced to make an action before interacting with a workspace.


https://github.com/user-attachments/assets/81c3c3bb-3c74-43c8-8820-4e96ae307f35

How do we fix this? I almost feel like `DeepLinksService` should always bail out to the "Connect" screen. As in, when `loginAndSetActiveWorkspace` is called, we check if the current workspace is equal to the workspace from the URL. If not, then we always zero `rootClusterUri` in `WorkspacesService` to force the user back to cluster choice. Otherwise, I feel it kind of makes little sense to launch the app through a link to workspace B, interact with a modal about workspace B (the login modal), then close it and be back at workspace A, just because it was the last used workspace before the app was closed.

I'm just not sure at this point how this is going to interact with the rest of the app.

Reply:
At first it seemed like a good idea, but then I realized that setting `rootClusterUri` to undefined unmounts the workspaces, so all the state kept in tabs will be lost :/ And since a deep link can be opened while you work on something, you would loose it.

Because of that, I decided to keep it as is.

Reply:
> At first it seemed like a good idea, but then I realized that setting `rootClusterUri` to undefined unmounts the workspaces, so all the state kept in tabs will be lost :/

That sounds like a behavior that we can change. But I'll look at the changes in this PR and the other PR first.